x-pack/plugins/reporting/server/lib/event_logger/logger.ts

```diff
@@ -27,171 +29,192 @@ export interface ExecutionCompleteMetrics {
   byteSize: number;
 }
 
+export interface IReportingEventLogger {
+  logEvent(message: string, properties: LogMeta): void;
+  startTiming(): void;
+  stopTiming(): void;
+}
+
+const getEventLog = (logger: LevelLogger) => ({
+  getLogger(lProperties: Partial<LogMeta>): IReportingEventLogger {
+    return (function () {
+      let start: Date | undefined;
+      let end: Date | undefined;
+      let duration: number | undefined;
```

Comment:
IIFE is not needed here. Since we create a new instance on every method call, these variables will be enclosed anyway.

Reply:
you are correct! 7d2ff62ae85